client/client.go

```diff
@@ -0,0 +1,91 @@
+package client
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrNotFound = errors.New("hardware not found")
+)
+
+// Finder is a type for discovering hardware.
+type Finder interface {
+	ByIP(context.Context, net.IP) (Discoverer, error)
+	ByMAC(context.Context, net.HardwareAddr, net.IP, string) (Discoverer, error)
+}
```

Comment:
Can you do `HardwareFinder` instead of bare `Finder` please, if we've got 2 of them then I think it makes most sense to have them easier to identify by type/name.

Reply:
Sure